docs/spec/v1beta2/imagerepositories.md

```diff
@@ -0,0 +1,610 @@
+
+In case of a CA certificate for the client to use, the data key for it is
+`caFile`. Adapting the previous example, if the certificate is in the file
+`ca.crt`, and the client certificate and key are as before, the whole command
+would be:
+
+```sh
+SECRET_NAME=tls-certs
+kubectl create secret generic $SECRET_NAME \
+  --from-file=certFile=client.crt \
+  --from-file=keyFile=client.key \
+  --from-file=caFile=ca.crt
+```
+
+### Suspend
+
+`.spec.suspend` is an optional field to suspend the reconciliation of an
+ImageRepository. When set to `true`, the controller will stop reconciling the
+ImageRepository, and changes to the resource or image repository will not result
+in new scan results. When the field is set to `false` or removed, it will
+resume.
+
+### Access from
+
+`.spec.accessFrom` is an optional field to restrict cross-namespace access of
+ImageRepositories. To grant access to an ImageRepository for policies in other
+namespaces, the owner of the ImageRepository has to specify a list of label
+selectors that match the namespace labels of the ImagePolicy objects.
+
+```yaml
+---
+apiVersion: image.toolkit.fluxcd.io/v1beta2
+kind: ImageRepository
+metadata:
+  name: app1
+  namespace: apps
+spec:
+  interval: 1h
+  image: docker.io/org/image
+  secretRef:
+    name: regcred
+  accessFrom:
+    namespaceSelectors:
+      - matchLabels:
+          kubernetes.io/metadata.name: flux-system
+```
+
+**Note** that the `kubernetes.io/metadata.name` is a readonly label added by
+Kubernetes >= 1.21 automatically on namespaces. For older version of Kubernetes,
+please set labels on the namespaces where the ImagePolicy exist.
+
+The above definition, allows or ImagePolicy in the `flux-system` namespace to
+reference the `app1` ImageRepository e.g.:
+
+```yaml
+---
+apiVersion: image.toolkit.fluxcd.io/v1beta2
+kind: ImagePolicy
+metadata:
+  name: app1
+  namespace: flux-system
+spec:
+  imageRepositoryRef:
+    name: app1
+    namespace: apps
+  policy:
+    semver:
+      range: 1.0.x
+```
+
+To grant access to all namespaces, an empty `matchLabels` can be set:
+
+```yaml
+  accessFrom:
+    namespaceSelectors:
+      - matchLabels: {}
+```
+
+### Exclusion list
+
+`.spec.exclusionList` is an optional field to exclude certain tags in the image
+scan result. It's a list of regular expression patterns with a default value of
+`"^.*\\.sig$"` if it's not set. This default value is used to exclude all the
+tags ending with `.sig`, since these are [Cosign](https://github.com/sigstore/cosign)
+generated objects and not container images which can be deployed on a Kubernetes
+cluster.
+
+```yaml
+---
+apiVersion: image.toolkit.fluxcd.io/v1beta2
+kind: ImageRepository
+metadata:
+  name: app1
+  namespace: apps
+spec:
+  interval: 1h
+  image: docker.io/org/image
+  exclusionList:
+    - "^.*\\.sig$"
+    - "1.0.2"
+    - "1.1.1|1.0.0"
+```
+
+## Working with ImageRepositories
+
+### Contextual Authentication
```

Comment:
I'm for holding off the `v1beta2` release until we add the `provider` field to the spec and we deprecate the controller flags. This is not blocking this PR but rather the release of the new API version.

Reply:
`provider` is now added in the ImageRepository API and the respective integration tests have been updated to test it against all the cloud providers. Ran the tests myself and they work.